### PR TITLE
Updates to enable live testing in sovereign clouds for multiple services

### PR DESCRIPTION
**Pull request by v-hongli1**
We are running live Tests against other clouds like US Gov and Azure China Cloud. The goal is to check whether new azure sdk package work with other clouds or not.
updated List: eventhubs/identity/storage

### CODE REVIEW

File: sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/EventHubAsyncClientIntegrationTest.java [hunk 1]
```diff
@@ -129,6 +129,12 @@ void getPropertiesWithCredentials() {
         // Arrange
 
         // Act & Assert
+        logger.info(this.getClass().getName()  + "getPropertiesWithCredentials " + "AZURE_EVENTHUBS_FULLY_QUALIFIED_DOMAIN_NAME : " + System.getenv("AZURE_EVENTHUBS_FULLY_QUALIFIED_DOMAIN_NAME"));
+        logger.info(this.getClass().getName()  + "getPropertiesWithCredentials " + "AZURE_EVENTHUBS_EVENT_HUB_NAME : " + System.getenv("AZURE_EVENTHUBS_EVENT_HUB_NAME"));
+        logger.info(this.getClass().getName()  + "getPropertiesWithCredentials " + "AZURE_CLIENT_ID : " + System.getenv("AZURE_CLIENT_ID"));
+        logger.info(this.getClass().getName()  + "getPropertiesWithCredentials " + "AZURE_CLIENT_SECRET : " + System.getenv("AZURE_CLIENT_SECRET"));
```
Comment (benbp) [inline sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/EventHubAsyncClientIntegrationTest.java:135]: We do not in any circumstances want to be logging secrets (here and a few places below).

Comment (hongli750210) [reply sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/EventHubAsyncClientIntegrationTest.java:135]: The above logging is only used to investigate the ci exception, and will be remove after the investigation.

Comment (benbp) [reply sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/EventHubAsyncClientIntegrationTest.java]: Meaning you intend to only run it locally? If it's ever run in CI, that secret is now exposed in the logs and needs to be rotated.

Comment (hongli750210) [reply sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/EventHubAsyncClientIntegrationTest.java]: Okey, The above Logging has been removed

Comment (benbp) [reply sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/EventHubAsyncClientIntegrationTest.java]: Thank you!

Comment (hongli750210) [reply sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/EventHubAsyncClientIntegrationTest.java]: That's welcome.

File: sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/BlobAPITest.groovy [hunk 2]
```diff
@@ -855,7 +855,7 @@ class BlobAPITest extends APISpec {
     def "Download file"() {
         setup:
         def file = getRandomFile(fileSize)
-        bc.uploadFromFile(file.toPath().toString(), true)
+        bc.uploadFromFile(file.toPath().toString(), new ParallelTransferOptions().setBlockSizeLong(4 * 1024 * 1024), null, null, null, null, null)
```
Comment (kasobol-msft) [inline sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/BlobAPITest.groovy:858]: what's the reason for this change?

File: sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/specialized/BlobBaseAPITest.groovy [hunk 3]
```diff
@@ -319,7 +319,7 @@ class BlobBaseAPITest extends APISpec {
         bc.uploadFromFile(f.getAbsolutePath(), true)
         byte[] expectedData = "0,mdifjt55.ea3,mdifjt55.ea3\n".getBytes()
 
-        def expression = "select * from blobstorage where id < 1;"
+        def expression = "select * from blobstorage where id &lt; 1;"
```
Comment (kasobol-msft) [inline sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/specialized/BlobBaseAPITest.groovy:322]: was this failing ?

File: sdk/storage/tests.yml [hunk 4]
```diff
@@ -4,6 +4,7 @@ stages:
   - template: /eng/pipelines/templates/stages/archetype-sdk-tests.yml
     parameters:
       ServiceDirectory: storage
+      SupportedClouds: 'Public,Canary,UsGov,China'
```
Comment (kasobol-msft) [inline sdk/storage/tests.yml:7]: 1. Can we please not add this to nightly runs? This will bring more flaky-ness. Instead I suggest to make it part of weekly pipe https://dev.azure.com/azure-sdk/internal/_build?definitionId=2714 where we already have few clouds listed. Regressions that are sovereign cloud specific are rare.
2. Storage tests are running on dedicated agent pool in canada central region where we provision test accounts (see line `Pool=(.*)-general/$1-storage` below). We heavily rely on that co-location to run tests with larger payloads. A mechanism to disable these tests in sovereign clouds should be developed first or we should get agents that are co-located there.

File: sdk/storage/test-resources.json [hunk 5]
```diff
@@ -21,6 +21,13 @@
...
                 "description": "The principal to assign the role to. This is application object id."
             }
+        },
+        "enableStorageVersioning": {
+          "type": "bool",
+          "metadata": {
+            "description": "Enable Versioning enabled setting, default value is true"
+          },
+          "defaultValue": true
         }
```
Comment (kasobol-msft) [inline sdk/storage/test-resources.json:31]: What's the reason for this? We already have `versionedAccountName` in the account set used in this arm template.
Some tests in the suite rely on versioning enabled and some on versioning enabled.

**Review by kasobol-msft — CHANGES_REQUESTED** [review]

File: sdk/storage/tests.yml [hunk 6]
```diff
@@ -61,8 +60,6 @@ stages:
             Path: sdk/storage/platform-matrix.json
             Selection: sparse
             GenerateVMJobs: true
-      PreSteps:
-          - template: /sdk/storage/tests-install-azurite.yml
```
Comment (kasobol-msft) [inline sdk/storage/tests.yml:65]: What's the replacement for this?

File: sdk/storage/tests.yml [hunk 7]
```diff
@@ -41,9 +42,7 @@ stages:
         - name: azure-storage-perf
           groupId: com.azure
       TestMode: 'LIVE'
-      TimeoutInMinutes: 60
-      Location: canadacentral
-      Clouds: 'Preview'
```
Comment (kasobol-msft) [inline sdk/storage/tests.yml:46]: We can't do that in nightly pipeline.

File: sdk/storage/tests.yml [hunk 8]
```diff
@@ -41,9 +42,7 @@ stages:
...
+      TimeoutInMinutes: 240
```
Comment (kasobol-msft) [inline sdk/storage/tests.yml:45]: If we're going beyond 60 minutes we should redesign the approach. It is also possible that you're running into many failures and retries which add up to large time. Normally live test suite for java completes below 30 minutes.

**Review by kasobol-msft — CHANGES_REQUESTED** [review]

File: sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/BlobAPITest.groovy (hunk 2 above)
Comment (hongli750210) [reply sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/BlobAPITest.groovy]: This change has been rolled back.

File: sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/specialized/BlobBaseAPITest.groovy (hunk 3 above)
Comment (hongli750210) [reply sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/specialized/BlobBaseAPITest.groovy]: This change has been rolled back

File: sdk/storage/tests.yml (hunk 4 above)
Comment (hongli750210) [reply sdk/storage/tests.yml]: This change has been rolled back

File: sdk/storage/test-resources.json (hunk 5 above)
Comment (hongli750210) [reply sdk/storage/test-resources.json]: This change has been rolled back

File: sdk/storage/tests.yml (hunk 6 above)
Comment (hongli750210) [reply sdk/storage/tests.yml]: This change has been rolled back

File: sdk/storage/tests.yml (hunk 7 above)
Comment (hongli750210) [reply sdk/storage/tests.yml]: This change has been rolled back

File: sdk/storage/tests.yml (hunk 8 above)
Comment (hongli750210) [reply sdk/storage/tests.yml]: This change has been rolled back

**Comment by hongli750210** [comment]
This PR contains too many commits, which affects the review of the code and the query of iterative records, so it is closed and replaced with [PR#23533](https://github.com/Azure/azure-sdk-for-java/pull/23533).
